### DISCUSSION TITLE
How to export gas trade share in GCAM 7.0

### DISCUSSION THREAD

**Discussion by whitenightZhang** [JGCRI/gcam-core#333] [Q&A] [answered]
The GCAM 7.0 model has a new natural gas trading module, and I was wondering how I should write a query to output the **SHARE** of the different natural gas import **subsectors (pipline and LNG)** for each of the regions in the regional natural gas?

**Comment by pkyle** [comment]
Technology and subsector shares aren't written to the output database; it's just quantities. Still the challenge with this query is that the regional natural gas sector has been structured with multiple levels of subsectors in order to get three levels of nesting competition. The upper nest is between domestic and imports, then within imports there's a level between LNG and pipeline gas, and then within pipeline there's the different regional trading blocks. Most regions only have access to one pipeline market and as such don't really need this second level, but some do (e.g. Central Asia has access to the Russian and Middle Eastern pipeline gas markets). Here's a query that collapses the subsectors, but still reports the necessary information to compute shares in post-processing.
`<supplyDemandQuery title="regional natural gas inputs">
    <axis1 name="input">input</axis1>
    <axis2 name="Year">demand-physical[@vintage]</axis2>
    <xPath buildList="true" dataName="input" group="false" sumAll="false">*[@type='sector' and (@name='regional natural gas')]//*[@type='technology']/*[@type='input']/demand-physical/node()</xPath>
    <comments/>
</supplyDemandQuery>
`

**Reply by whitenightZhang** [reply]
Sorry, I was trying to output the **share-weight** of the subsector, please forgive me for expressing it incorrectly

**Reply by pkyle** [reply, marked as answer]
Well that changes things! Since there are two levels of subsectors, one of these will work:
`<supplyDemandQuery title="regional natural gas subsector share-weights">
    <axis1 name="subsector">subsector</axis1>
    <axis2 name="Year">share-weight</axis2>
    <xPath buildList="true" dataName="price" group="false" sumAll="false">*[@type='sector' and (@name='regional natural gas')]/
               *[@type='subsector']/
               share-weight/text()</xPath>
    <comments/>
</supplyDemandQuery>
`
Or
`<supplyDemandQuery title="regional natural gas sub-subsector share-weights">
    <axis1 name="subsector">subsector</axis1>
    <axis2 name="Year">share-weight</axis2>
    <xPath buildList="true" dataName="price" group="false" sumAll="false">*[@type='sector' and (@name='regional natural gas')]/
               *[@type='subsector']/*[@type='subsector']/
               share-weight/text()</xPath>
    <comments/>
</supplyDemandQuery>
`

**Reply by whitenightZhang** [reply]
Thanks, that's very helpful!

**Comment by pkyle** [comment]
Also here is the core model proposal documenting the structure in case it's helpful: https://jgcri.github.io/gcam-doc/cmp/350-Detailed_Natural_Trade.pdf
